runtime/dancebox/tests/integration_test.rs

```diff
@@ -2307,6 +2307,80 @@ fn test_proxy_non_transfer() {
         });
 }
 
+#[test]
+fn test_proxy_utility() {
+    // All proxy types should be able to use Utility pallet, but we ensure
+    // subcalls don't allow to circumvent filters.
+
+    // Dummy match to ensure we update this test when adding new proxy types.
+    match ProxyType::Any {
+        ProxyType::Any
+        | ProxyType::NonTransfer
+        | ProxyType::Governance
+        | ProxyType::Staking
+        | ProxyType::CancelProxy
+        | ProxyType::Balances
+        | ProxyType::Registrar
+        | ProxyType::SudoRegistrar => (),
+    };
+
+    let proxy_types = &[
+        ProxyType::Any,
+        ProxyType::NonTransfer,
+        ProxyType::Governance,
+        ProxyType::Staking,
+        ProxyType::CancelProxy,
+        ProxyType::Balances,
+        ProxyType::Registrar,
+        ProxyType::SudoRegistrar,
+    ];
```

Comment:
There are some external crates that allow iterating over enums, not sure if we already have one, but as a nice hack we could use the Decode trait (not tested but should work):

```
for x in 0u8.. {
    if let Ok(x) = x.decode::<ProxyType>() {
        proxy_types.push(x);
    } else {
        break;
    }
}
```